twoliter/embedded/imghelper

```diff
@@ -1,6 +1,150 @@
 #!/usr/bin/env bash
-
 # shellcheck disable=SC2034 # variables are used through parameter substitution
+
+DATA_IMAGE_SUFFIX="-data"
+BOOT_IMAGE_SUFFIX="-boot.ext4.lz4"
+VERITY_IMAGE_SUFFIX="-root.verity.lz4"
+ROOT_IMAGE_SUFFIX="-root.ext4.lz4"
+
+set_image_names() {
+  local filename_prefix
+  local -n os_image_name data_image_name
+  local -n boot_image_name verity_image_name root_image_name
```

Comment:
I don't see why the caller needs to have these three names (boot/root/hash). `rpm2img` never interacts with files by these names because they're only created when compressed.

Also, `rpm2img` only needs `os_image_name` and `data_image_name` because the OVA creation change hasn't merged yet.

I would say, try to merge that one first, rebase, and remove the `set_image_names` and `get_data_image` functions until the caller is added, so I have more context to review.

Reply:
Once everything is moved over, only `imghelper` will use them, but it felt kind of wrong to have `FILENAME_PREFIX="${IMAGE_NAME}-${VARIANT}-${ARCH}-${VERSION_ID}-${BUILD_ID}"` there. These values come from the docker environment calling `rpm2img` or `img2img`, but if we're ok with FILENAME_PREFIX (and the similar SYMLINK_PREFIX) being set at the top of `imghelper` it would make the code cleaner on both sides.

Reply:
> if we're ok with FILENAME_PREFIX (and the similar SYMLINK_PREFIX) being set at the top of `imghelper` it would make the code cleaner on both sides.

OK, let's try it like that.